test/limitOrderBook/limitOrderBook.fillOrder.test.ts

```diff
@@ -0,0 +1,327 @@
+            // long 0.1 ETH at $3000 with $300
+            const limitOrder = {
+                salt: 1,
+                trader: trader.address,
+                baseToken: baseToken.address,
+                isBaseToQuote: false,
+                isExactInput: true,
+                amount: parseEther("300"),
+                oppositeAmountBound: parseEther("0.1"),
+                deadline: ethers.constants.MaxUint256,
+                reduceOnly: true,
+            }
+
+            // sign limit order
+            const signature = await getSignature(fixture, limitOrder, trader)
+
+            await expect(limitOrderBook.connect(keeper).fillLimitOrder(limitOrder, signature)).to.be.revertedWith(
+                "LOB_TINR",
+            )
+        })
+
+        it("force error, when order does not satisfy reduceOnly, create a reverse position", async () => {
+            // short 0.2 ETH at $2800 with $560
+            const limitOrder = {
+                salt: 1,
+                trader: trader.address,
+                baseToken: baseToken.address,
+                isBaseToQuote: true,
+                isExactInput: true,
+                amount: parseEther("0.2").toString(),
+                oppositeAmountBound: parseEther("560").toString(),
+                deadline: ethers.constants.MaxUint256.toString(),
+                reduceOnly: true,
+            }
+
+            // sign limit order
+            const signature = await getSignature(fixture, limitOrder, trader)
+
+            await expect(limitOrderBook.connect(keeper).fillLimitOrder(limitOrder, signature)).to.be.revertedWith(
+                "LOB_TINR",
+            )
+        })
+    })
+
+    it("cancel order successfully", async () => {
+        // long 0.1 ETH at $3000 with $300
+        const limitOrder = {
+            salt: 1,
+            trader: trader.address,
+            baseToken: baseToken.address,
+            isBaseToQuote: false,
+            isExactInput: true,
+            amount: parseEther("300"),
+            oppositeAmountBound: parseEther("0.1"),
+            deadline: ethers.constants.MaxUint256,
+            reduceOnly: false,
+        }
+
+        await expect(await limitOrderBook.connect(trader).cancelLimitOrder(limitOrder)).to.emit(
+            limitOrderBook,
+            "LimitOrderCancelled",
+        )
+    })
+
+    it("force error, cancel order with different trader", async () => {
+        // long 0.1 ETH at $3000 with $300
+        const limitOrder = {
+            salt: 1,
+            trader: keeper.address,
+            baseToken: baseToken.address,
+            isBaseToQuote: false,
+            isExactInput: true,
+            amount: parseEther("300"),
+            oppositeAmountBound: parseEther("0.1"),
+            deadline: ethers.constants.MaxUint256,
+            reduceOnly: false,
+        }
+
+        await expect(limitOrderBook.connect(trader).cancelLimitOrder(limitOrder)).to.be.revertedWith("LOB_OSMBS")
+    })
+})
```

Comment:
Might be better to have some failed cases for deadline or price/amount is not matched.